### PR TITLE
Simplify version parsing in updating workflow

### PR DESCRIPTION
**Pull request by charlesbluca**
As https://github.com/rapidsai/dask-build-environment/pull/31 shows, the auto-updating workflow can stop working properly if minor changes are made to the axis YAML or build script. This is because the parsing of the old/current RAPIDS / UCX-Py versions is done using a combination of `grep` and `tail`, which assumes a very specific structure to the files being parsed.

This PR attempts to resolve this issue by generalizing the process of parsing these versions. Now, they are parsed as follows:

- the old / current RAPIDS versions are obtained by parsing the YAML file and specifically grabbing the values of `RAPIDS_VER`
- the old / current ucx-py versions are obtained by passing the obtained RAPIDS versions to [RVC](https://github.com/rapidsai/rvc/#aws-lambda-instance)

This should hopefully improve the stability of this workflow moving forward.

### CODE REVIEW

File: .github/workflows/update-gpuci.yml [hunk 1]
```diff
@@ -13,12 +13,17 @@ jobs:
...
+      - name: Get old RAPIDS / UCX-Py versions
         run: |
-          echo OLD_RAPIDS_VER=$(grep -A 1 "RAPIDS_VER" ci/axis/dask.yaml | tail -n1 | grep -o "'.*'" | sed "s/'//g") >> $GITHUB_ENV
-          echo OLD_UCX_PY_VER=$(grep -m1 "UCX_PY_VER" ci/gpuci/run.sh | tail -n1 | grep -o '".*"' | sed 's/"//g') >> $GITHUB_ENV
-          echo RAPIDS_VER=$(grep -A 2 "RAPIDS_VER" ci/axis/dask.yaml | tail -n1 | grep -o "'.*'" | sed "s/'//g") >> $GITHUB_ENV
-          echo UCX_PY_VER=$(grep -m2 "UCX_PY_VER" ci/gpuci/run.sh | tail -n1 | grep -o '".*"' | sed 's/"//g') >> $GITHUB_ENV
+          echo OLD_RAPIDS_VER=$RAPIDS_VER_0 >> $GITHUB_ENV
+          echo OLD_UCX_PY_VER=$(curl -sL https://version.gpuci.io/rapids/$RAPIDS_VER_0) >> $GITHUB_ENV
+          echo RAPIDS_VER=$RAPIDS_VER_1 >> $GITHUB_ENV
+          echo UCX_PY_VER=$(curl -sL https://version.gpuci.io/rapids/$RAPIDS_VER_1) >> $GITHUB_ENV
```
Comment (pentschev) [inline .github/workflows/update-gpuci.yml:26]: From these changes it's not clear to me where do `RAPIDS_VER_0`/`RAPIDS_VER_1` come from. Are those reliable? Is there any chance that 0 and 1 may be swapped elsewhere and cause similar problems in the future?

**Review by pentschev — APPROVED** [review]
Thanks @charlesbluca , overall looks good to me. I have only a question about variables containing versions, but if you think those are fine we can go ahead and merge it.

File: .github/workflows/update-gpuci.yml (hunk 1 above)
Comment (charlesbluca) [reply .github/workflows/update-gpuci.yml:26]: After parsing the axis YAML using this [YAML parsing Action](https://github.com/marketplace/actions/yaml-parser), all of the keys in the YAML get dumped into the GitHub Actions environment, using `_` as a delimiter and appending integers to indicate ordering in lists.

Thus, `RAPIDS_VER_0` / `RAPIDS_VER_1` represent the first and second elements under the `RAPIDS_VER` key in the axis YAML:

https://github.com/rapidsai/dask-build-environment/blob/210c6be6249322f1ac022cd01ab6335c29fa4e2c/ci/axis/dask.yaml#L17-L19

As long as we don't manually change the order of these elements, this ordering should stay consistent.

Comment (pentschev) [reply .github/workflows/update-gpuci.yml:26]: Ok, but this means changes from https://github.com/rapidsai/dask-build-environment/pull/31 are breaking this and must be fixed, right?

Comment (charlesbluca) [reply .github/workflows/update-gpuci.yml:26]: Yeah, the updates made in #31 are erroneous and my intention after merging this is to close that PR and rerun this updated workflow to make the proper changes

**Pull request merged**